packages/protocol/migrations/08_stabletoken.ts

```diff
@@ -37,31 +39,56 @@ module.exports = deploymentForCoreContract<StableTokenInstance>(
   artifacts,
   CeloContractName.StableToken,
   initializeArgs,
-  async (stableToken: StableTokenInstance, _web3: Web3, networkName: string) => {
-    const minerAddress: string = truffle.networks[networkName].from
-    console.log('Setting GoldToken/USD exchange rate')
+  async (stableToken: StableTokenInstance, web3: Web3, networkName: string) => {
     const sortedOracles: SortedOraclesInstance = await getDeployedProxiedContract<
       SortedOraclesInstance
     >('SortedOracles', artifacts)
 
-    for (const oracle of config.stableToken.oracles) {
+    console.info('Adding inital C Labs oracles to the SortedOracles whitelist')
+    const oracleAddresses = config.stableToken.oracleKeys.map(privateKeyToAddress)
+    for (const oracle of oracleAddresses) {
       console.info(`Adding ${oracle} as an Oracle for StableToken`)
       await sortedOracles.addOracle(stableToken.address, oracle)
     }
 
-    // We need to seed the exchange rate, and that must be done with an account
-    // that's accessible to the migrations. It's in an if statement in case this
-    // account happened to be included in config.stableToken.oracles
-    if (!(await sortedOracles.isOracle(stableToken.address, minerAddress))) {
+    // If there were no oracles set, we still need to set the initial exchange rate.
+    // Default to using the first validator
+    if (oracleAddresses.length === 0) {
+      console.info('No oracle keys were found in the config. Using the first validator instead')
+
+      const minerAddress: string = truffle.networks[networkName].from
+      console.info(`Adding ${minerAddress} as an Oracle for StableToken`)
       await sortedOracles.addOracle(stableToken.address, minerAddress)
+
+      console.info(
+        'Setting the initial GoldToken/USD exchange rate, reporting as the first validator'
+      )
+      await sortedOracles.report(
+        stableToken.address,
+        config.stableToken.goldPrice,
+        1,
+        NULL_ADDRESS,
+        NULL_ADDRESS
+      )
+    } else {
+      console.info(
+        `Setting the initial GoldToken/USD exchange rate, reporting as the first oracle: ${
+          oracleAddresses[0]
+        }`
+      )
+
+      // @ts-ignore
+      const reportTx = sortedOracles.contract.methods.report(
+        stableToken.address,
+        config.stableToken.goldPrice,
+        1,
+        NULL_ADDRESS,
+        NULL_ADDRESS
+      )
+      await sendTransactionWithPrivateKey(web3, reportTx, config.stableToken.oracleKeys[0], {
+        to: sortedOracles.address,
+      })
```

Comment:
This is the part that fails. This is the error, thrown inside of `sendTransactionWithPrivateKey`:
```
eth_sendRawTransaction
Error: Error: Error: Returned error: wrong number of fields in data
    at Object.run (/Users/audreypenven/celo/celo-monorepo/node_modules/truffle/build/webpack:/packages/truffle-migrate/index.js:92:1)
```

The same exact transaction sent differently above, from the default `from` on lines 66-72 above works without error.

Additionally, this code here looks identical in format to what's happening many times the later validator election migration. For example: https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/migrations/20_elect_validators.ts#L133-L137

So, I'm baffled as to why this is failing.

Reply:
Weird, looks like it's defined at https://github.com/ethereumjs/ethereumjs-util/blob/master/src/object.ts#L100

Could you maybe print out what `data` and `self._fields` is for you? Looks like some error in serialization maybe?